polyply/src/map_to_molecule.py

```diff
@@ -52,48 +55,41 @@ def expand_meta_graph(meta_molecule, block, meta_mol_node):
         residues are added instead of the orignial block to
         the meta molecule.
         """
-        # 1. relable nodes to make space for new nodes to be inserted
+        # 0. make edges for the block
+        for inter_type in ["bonds", "constraints", "virtual_sitesn",
+                           "virtual_sites2", "virtual_sites3", "virtual_sites4"]:
+            block.make_edges_from_interaction_type(inter_type)
+
+        # 1. make residue graph and check it is not branched
+        # we cannot do branched block expansion
+        expanded_graph = make_residue_graph(block)
+        if is_branched(expanded_graph):
+            msg = ("Block {} consists of more than one residue, however, "
+                   "it is branched. Cannot expand a branched block, because "
+                   "of lacking information on how the block connects to the "
+                   "rest.")
+            raise IOError(msg.format(block.name))
```

Comment:
I'm not sure I understand why this limitation exists.
If I understand correctly:
- You have blocks in your metaml that consist of multiple residues, so you expand those to separate nodes. Will those nodes/residues be connected to eachother, because you do `make_edges_from_interaction_type`?
- You check for branching
- You error because you don't know how to connect your new nodes to the existing molecule. However, the docstring of this class claims it makes disconnected Molecules, so I'd argue it's not the problem of this class that you don't know (yet) how to connect everything?

At the very least, requires an extensive comment (pref. with an example) that describes the issue.

Reply:
Almost, but I check the residue graph for it's branching. Basically this class generates a disconnected molecule not meta_molecule. The complete graph for the meta_molecule needs to be present. If you have a mutliple block single residue and you expand it, you only know when it is linear that the first and last residues are connected on the meta-molecule level. Does that make sense?

Reply:
Nope, not yet I'm afraid.
Do you have an example?